tlsn/tlsn-notary-client/src/client.rs

```diff
@@ -1,29 +1,40 @@
 //! Notary client
 //!
-//! This module sets up prover to connect to notary via TCP or TLS
+//! This module sets up connection to notary server via TCP or TLS, and subsequent requests for notarization
 
-use eyre::eyre;
 use http_body_util::{BodyExt as _, Either, Empty, Full};
 use hyper::{client::conn::http1::Parts, Request, StatusCode};
 use hyper_util::rt::TokioIo;
 use notary_server::{ClientType, NotarizationSessionRequest, NotarizationSessionResponse};
-use rustls::{Certificate, ClientConfig, RootCertStore};
 use std::sync::Arc;
-use tls_client::RootCertStore as TlsClientRootCertStore;
+use tls_client::{ClientConfig, ClientConnection, RootCertStore, RustCryptoBackend, ServerName};
+use tls_client_async::{bind_client, TlsConnection};
 use tlsn_common::config::{DEFAULT_MAX_RECV_LIMIT, DEFAULT_MAX_SENT_LIMIT};
-use tlsn_prover::tls::{state::Setup, Prover, ProverConfig};
-use tokio::io::{AsyncRead, AsyncWrite};
-use tokio_rustls::TlsConnector;
-use tokio_util::{bytes::Bytes, compat::TokioAsyncReadCompatExt};
+use tokio::{
+    io::{AsyncRead, AsyncWrite},
+    net::TcpStream,
+};
+use tokio_util::{
+    bytes::Bytes,
+    compat::{Compat, FuturesAsyncReadCompatExt, TokioAsyncReadCompatExt},
+};
 
 #[cfg(feature = "tracing")]
 use tracing::debug;
 
-use crate::error::NotaryClientError;
+use crate::error::{ClientError, ErrorKind};
 
-/// Client that setup prover to connect to notary server
+/// A Notary connection
+pub enum NotaryConnection {
+    /// Unencrypted TCP connection
+    Tcp(TcpStream),
+    /// TLS connection wrapped in Compat that implement tokio's AsyncRead, AsyncWrite
+    Tls(Compat<TlsConnection>),
```

Comment:
Was trying to use our modified one so that user can just use a single RootCertStore type for both server and notary's TLS setup, instead of a different RootCertStore (e.g. rustls's) just for notary (prover's config is using tls-client's [RootCertStore](https://github.com/tlsnotary/tlsn/blob/main/tlsn/tlsn-prover/src/tls/config.rs#L21)) — but maybe that'll be kinda edge case since most people ll use the default webpki and not specify any? What do you think?

Reply:
Decision: to use rustls' RootCertStore as most people would just use our default